syntax = "proto3";

option objc_class_prefix = "RTG";

package social_media;

service PostsService {
  rpc GetPost(GetPostRequest) returns (GetPostResponse) {}
  rpc GetPostComments(GetPostCommentsRequest) returns (GetPostCommentsResponse) {}
}

message GetPostRequest {
  int32 post_id = 1;
}

message GetPostResponse {
  bool found = 1;
  optional Post post = 2;
}

message Post {
  int32 post_id = 1;
  int32 user_id = 2;
  string content = 3;
  uint64 post_date_timestamp = 4;
}

message GetPostCommentsRequest {
  int32 post_id = 1;
}

message GetPostCommentResponse {
  int32 post_id = 1;
  int32 comment_id = 2;
  int32 author_id = 3;
  string content = 4;
  uint64 timestamp = 5;
  string timestamp_relative = 6;
  int32 comment_likes_count = 7;
}

message GetPostCommentsResponse {
  int32 count = 1 ;
  repeated GetPostCommentResponse items = 2;
}
